### PR TITLE
Feature/soca remove cdate

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
Removes references to CDATE as non-local variable; in a couple of instances it is assigned locally to `${PDY}${cyc}`. Addresses issue https://github.com/NOAA-EMC/GDASApp/issues/237

Tested with ctests+ in build/test (ie, not just for soca)

Depends on the merging of global-workflow PR https://github.com/NOAA-EMC/global-workflow/pull/1191 to work for g-w, but  should work for soca alone

Files changed:
parm/soca/obs/config/adt_j3.yaml
parm/soca/obs/config/icec_emc.yaml
parm/soca/obs/config/salt_profile_fnmoc.yaml
parm/soca/obs/config/sss_smap.yaml
parm/soca/obs/config/sst_noaa19_l3u.yaml
parm/soca/obs/config/sst_noaa19_l3u_so025.yaml
scripts/exgdas_global_marine_analysis_prep.py
test/soca/gw/config.base
test/soca/gw/runtime_vars.sh
test/soca/runtime_vars.sh
ush/ufsda/misc_utils.py

### CODE REVIEW

File: scripts/exgdas_global_marine_analysis_prep.py
Comment (CoryMartin-NOAA) [inline scripts/exgdas_global_marine_analysis_prep.py]: I know this is @guillaumevernieres 's problem, not yours, but `$COMOUT` should not have `ocnanal_`... in it. Just making a note here.

File: test/soca/gw/config.base
Comment (CoryMartin-NOAA) [inline test/soca/gw/config.base:98]: if you don't export it, it only works in this file

File: test/soca/runtime_vars.sh
Comment (CoryMartin-NOAA) [inline test/soca/runtime_vars.sh]: See my comment in g-w about this path

File: test/soca/gw/config.base
Comment (AndrewEichmann-NOAA) [reply test/soca/gw/config.base:98]: Isn't that the idea?

Comment (CoryMartin-NOAA) [reply test/soca/gw/config.base:98]: fair enough

**Comment by guillaumevernieres** [comment]
@AndrewEichmann-NOAA , the `CI` won't work until your gw PR is merged.

File: scripts/exgdas_global_marine_analysis_prep.py
Comment (guillaumevernieres) [reply scripts/exgdas_global_marine_analysis_prep.py]: Not sure I understand your comment @CoryMartin-NOAA `$COMOUT` doesn't have `ocnanal_...` in it ... Or do you mean anl_out should be something else?

_I'm being generous and decided to share some of my problems with @AndrewEichmann-NOAA_

Comment (CoryMartin-NOAA) [reply scripts/exgdas_global_marine_analysis_prep.py]: if you look at like the existing atmos GDAS ROTDIR, there is no subdirectory for analysis stuff, it is all there together. Not sure this is a 'requirement' or not, just thinking of consistency.

Comment (guillaumevernieres) [reply scripts/exgdas_global_marine_analysis_prep.py]: OK, saved as an issue @CoryMartin-NOAA .

**Review by guillaumevernieres — COMMENTED** [review]
The CI worked @AndrewEichmann-NOAA  but you need to address @CoryMartin-NOAA 's comment about the missing `$RUN` as a prefix. Looks good otherwise.

**Comment by guillaumevernieres** [comment]
I'm taking the liberty to push to your branch @AndrewEichmann-NOAA ... Needs a few minor updates.

File: test/soca/runtime_vars.sh
Comment (guillaumevernieres) [reply test/soca/runtime_vars.sh]: We have an issue to address the extra dir level, if that's what you were reffering to @CoryMartin-NOAA , I removed PDY from the name if that's what your comment was about.

**Review by RussTreadon-NOAA — APPROVED** [review]
Changes look straightforward.  Approve pending successful GW-RT on Hera and Orion.

**Comment by AndrewEichmann-NOAA** [comment]
For clarification, is this the comment in https://github.com/NOAA-EMC/global-workflow/pull/1191 that needs to be addressed?

[CoryMartin-NOAA](https://github.com/CoryMartin-NOAA) (https://github.com/NOAA-EMC/global-workflow/pull/1191#discussion_r1051181907)
I think @aerorahul wants instead this to be `${RUN}ocnanal_${cyc}` per a conversation he and I had yesterday afternoon

Member
@[aerorahul](https://github.com/aerorahul) aerorahul (https://github.com/NOAA-EMC/global-workflow/pull/1191#discussion_r1051188752)
Yes. Thanks @CoryMartin-NOAA.
`${PDY}` is not available at this time, so the issue is correct on that, that `${CDATE}` should become `${cyc}`.

I also realized yesterday, that our run directories will get clobbered when gdas and gfs cycles start. To distinguish between those, it is best to prepend ocnanal with a `${RUN}`.

`${RUN} `is set upstream of the job in the ecflow/rocoto job-card. EMC internally translates that to `${CDUMP}`, which is also not available at this point in the j-job.

**Comment by guillaumevernieres** [comment]
I forgot if I added `RUN` yesterday @AndrewEichmann-NOAA , you might need to issue a bugfix.

**Comment by aerorahul** [comment]
@AndrewEichmann-NOAA @guillaumevernieres 
I don't think so.  This was merged yesterday and it looks right in `develop`.
```
export DATA=${DATA:-${DATAROOT}/${RUN}ocnanal_${cyc}}
```
I think the comment has been addressed.  Thank you for doing so.

**Pull request merged**
